rfcs/0011-project-structure.md

```diff
@@ -0,0 +1,116 @@
+
+# Proposed Implementation
+[proposed-implementation]: #proposed-implementation
+
+The current structure of a LF project is organized as follows:
+
+```shell
+├── root
+│   ├── bin/       # Binary executables
+│   ├── include/   # Header files
+│   ├── src/       # LF programs
+│   ├── src-gen/   # Generated code for local LF programs
+│   └── fed-gen/   # Generated code for federated programs (instead of src-gen)
+```
+
+To enhance the project structure, the following additions are proposed:
+
+- **lib/**: Directory for storing reusable reactors.
+- **Lingo.toml**: Configuration file crucial for project sharing, storing Lingo configuration details including external LF projects to include (refer to [Lingo Documentation](https://github.com/lf-lang/lingo?tab=readme-ov-file#the-toml-based-package-configurations) for further details).
+
+The refined structure will appear as follows:
+```shell
+├── root
+│   ├── bin/
+│   ├── include/
+│   ├── lib/ # Directory for storing reusable reactors
+│   │   ├── Input.lf # Ex: reactor capturing external inputs (e.g., Microphone, Camera)
+│   │   └── ComputerVision.lf # Ex: reactor performing computer vision tasks (e.g., object detection, face recognition)
+│   ├── src/
+│   ├── src-gen/
+│   ├── fed-gen/              # Only for federated programs (instead of src-gen)
+│   └── Lingo.toml            # Configuration file for Lingo Package Manager
+```
+
+When executing `lingo build`, additional directories will be created in the project root:
+
+```shell
+├── root
+│   ...
+│   ├── target/
+│   │   ├── lfc_include/      # Includes all downloaded external LF projects
+│   │   │   ├── remote_lf_project_1/
+│   │   │   └── remote_lf_project_2/
+│   │   ├── libraries/
+│   │   ├── library/
+│   ...
+```
+
+The ‘Lingua Franca Package Explorer’ in the VS Code extension demonstrates how to utilize this structure. It presents a TreeView with two sections:
+- **Local Libraries**: Lists libraries defined by the programmer, located at `root/lib/*.lf`.
+- **Lingo Libraries**: Lists libraries downloaded from the Lingo Package Manager, located at `root/target/lfc_include/remote_lf_project_1/lib/*.lf`.
+
+This structured approach promotes efficient code reuse, simplifies integration of external resources, and fosters collaborative development practices within the LF community.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Feel free to suggest potential drowbacks of the proposed strucuture.
```

Comment:
I think lack of flexibility is a drawback. We impose the structure on the user. This could be addressed later when we have robust lingo support by allowing the user to adjust aspects of the project structure in the configuration.

Reply:
In the meantime, I’ll add ‘lack of flexibility’ as a drawback